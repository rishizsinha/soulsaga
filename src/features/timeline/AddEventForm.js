import AddIcon from '@material-ui/icons/Add';
import AppBar from '@material-ui/core/AppBar';
import DatePicker from './DatePicker';
import Button from '@material-ui/core/Button';
import CloseIcon from '@material-ui/icons/Close';
import Cropper from 'react-cropper';
import Dialog from '@material-ui/core/Dialog';
import Divider from '@material-ui/core/Divider';
import EventProto from '../../proto/EventProto.js'
import FormControl from '@material-ui/core/FormControl';
import FormHelperText from '@material-ui/core/FormHelperText';
import IconButton from '@material-ui/core/IconButton';
import Input from '@material-ui/core/Input';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import MenuItem from '@material-ui/core/MenuItem';
import PropTypes from 'prop-types';
import React, { Component } from 'react';
import Slide from '@material-ui/core/Slide';
import Snackbar from '@material-ui/core/Snackbar';
import TagInput from './TagInput';
import TextField from '@material-ui/core/TextField';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import { PRIMARY_CATEGORIES, SECONDARY_CATEGORIES } from './eventiconmaps';
import { 
  dataOperationSnackbarFailed,
  dataOperationSnackbarSucceeded } from 'features/common/redux/actions';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import * as actions from './redux/actions';
import 'cropperjs/dist/cropper.css';
const _ = require('lodash');

function Transition(props) {
  return <Slide direction="up" {...props} />;
}

export class AddEventForm extends Component {
  static propTypes = {
    timeline: PropTypes.object.isRequired,
    actions: PropTypes.object.isRequired,
  };

  initState = {
    eventCategory: "Other",
    eventTitle: "",
    eventDesc: "",
    eventTags: [],
    eventMonth: "",
    eventDay: "",
    eventYear: "",
    eventImg: "",
    eventId: "",
    hasTitle: false,
    validDate: false,
    snackbarOpen: false,
  };

  state = {...this.initState};

  /** DIALOG VISIBILITY EVENTS */

  handleClickOpen = () => {
    this.props.actions.toggleAddEventForm(true);
  };

  refreshFields = () => {
    if (this.props.timeline.editingEvent != null) {
      const e = this.props.timeline.editingEvent;
      this.setState({
        eventCategory: e.c,
        eventTitle: e.t,
        eventDesc: e.de,
        eventTags: e.tg.split(','),
        eventMonth: e.m,
        eventDay: e.d,
        eventYear: e.y,
        eventImg: e.i,
        eventId: e.id,
        hasTitle: true,
        validDate: true,
      })
    }
  }

  handleClose = () => {
    this.setState({...this.initState});
    this.props.actions.toggleAddEventForm(false);
    this.props.actions.setEditingEvent(/* null */);
  };

  /** FORM MODIFICATION EVENTS */

  handleEventCategoryChange = e => {
    this.setState({eventCategory: e.target.value});
  }

  handleEventTitleChange = e => {
    this.setState({eventTitle: e.target.value, hasTitle: e.target.value !== ""});
  }

  handleEventDateChange = (m, d, y, valid) => {
    this.setState({
      eventMonth: m,
      eventDay: d,
      eventYear: y,
      validDate: valid,
    });
  }

  handleEventDescChange = e => {
    this.setState({eventDesc: e.target.value});
  }

  handleEventTagChange = tags => {
    this.setState({ eventTags: tags });
  }

  handleEventImgChange = e => {
    const reader = new FileReader();
    reader.addEventListener('load', () => this.setState({eventImg: reader.result}), false);
    if (e.target.files[0]) {
      reader.readAsDataURL(e.target.files[0]);
    } else {
      this.setState({eventImg: ""});
    }
  }

  handleSnackbarClose = (event, reason) => {
    if (reason === 'clickaway') {
      return;
    }
    this.setState({snackbarOpen: false});
  }

  submitEvent = () => {
    if (!this.state.hasTitle || !this.state.validDate) {
      this.setState({snackbarOpen: true});
    } else {
      // Delete editingEvent if exists
      if (this.props.timeline.editingEvent != null) {
        this.props.actions.deleteEvent(
          this.props.timeline.editingEvent.id,
          this.props.timeline.editingEvent.ref);
      }
      // (Re)add current event
      this.props.actions.addEvent(this.buildEventProto())
        .then(() => this.props.actions.dataOperationSnackbarSucceeded())
        .catch(() => this.props.actions.dataOperationSnackbarFailed());
      this.handleClose();
    }
  }

  /**
   * Build event proto from form info.
   * Reuses ID of editingEvent if exists.
   */
  buildEventProto = () => {
    const img = this.refs.cropper != null
      ? this.refs.cropper.getCroppedCanvas().toDataURL()
      : "";
    const id = this.props.timeline.editingEvent == null
      ? Date.now()
      : this.props.timeline.editingEvent.id;

    const e = new EventProto();
    e.c = this.state.eventCategory;
    e.t = this.state.eventTitle;
    e.de = this.state.eventDesc;
    e.tg = this.state.eventTags.join(',');
    e.y = this.state.eventYear.replace(/^[0]+/g,"");
    e.m = this.state.eventMonth;
    e.d = this.state.eventDay.replace(/^[0]+/g,"");
    e.i = img;
    e.id = id;
    e.ms = this.buildDateTime(e.d, e.m, e.y);
    return e
  }

  buildDateTime = (d, m, y) => {
    const MONTHS = { 
      "January" : 1, "February" : 2, "March" : 3, "April" : 4, "May" : 5,
      "June" : 6, "July" : 7, "August" : 8, "September" : 9, "October" : 10,
      "November" : 11, "December" : 12
    }
    var date = new Date();
    if (d === "" && m === "") {
      date.setFullYear(parseInt(y, 10)+1);
      date.setMonth(0);
      date.setDate(0);
      date.setHours(23);
      date.setMinutes(59);
    } else if (d === "") {
      date.setFullYear(parseInt(y, 10));
      date.setMonth(MONTHS[m])
      date.setDate(0);
      date.setHours(23);
      date.setMinutes(0);
    } else {
      date.setFullYear(parseInt(y, 10));
      date.setMonth(MONTHS[m]-1);
      date.setDate(parseInt(d, 10));
      date.setHours(1);
      date.setMinutes(0);
    }
    return date.getTime();
  }

  renderCropper = () => {
    if (this.state.eventImg) {
      return (
        <Cropper
            ref="cropper"
            src={this.state.eventImg}
            style={{width: 640, height: 360}}
            aspectRatio={16 / 9}
            autoCropArea={1}
            guides={true}
            dragMode="move"/>
      );
    }
  }

  render() {
    const titleError = this.state.eventTitle === "";
    return (
      <div>
        {/** Floating '+' action button */}
        <Button
            variant="fab"
            color="primary"
            className="timeline-add-event-fab"
            onClick={this.handleClickOpen}>
          <AddIcon />
        </Button>

        {/** Fullscreen add event dialog */}
        <Dialog
          fullScreen
          open={this.props.timeline.isAddEventFormOpen}
          onEnter={this.refreshFields}
          onClose={this.handleClose}
          TransitionComponent={Transition}
          aria-labelledby="form-dialog-title"
        >
          <AppBar className="timeline-add-event-dialog__app-bar">
            <Toolbar>
              <IconButton color="inherit" onClick={this.handleClose} aria-label="Close">
                <CloseIcon />
              </IconButton>
              <Typography variant="title" color="inherit" className="timeline-add-event-dialog__app-bar-text">
                Add Event
              </Typography>
              <Button color="inherit" onClick={this.submitEvent}>
                Save
              </Button>
            </Toolbar>
          </AppBar>
          
          <form className="timeline-add-event-form__container">

            {/** Event Category Select */}
            <TextField
              select
              label="Category"
              value={this.state.eventCategory}
              onChange={this.handleEventCategoryChange}
              className="timeline-add-event__category-field"
              margin="normal"
              SelectProps={{
                MenuProps: {
                  className: "timeline-add-event__category-select-menu",
                },
              }}
              InputProps={{
                classes: {
                  input: "timeline-add-event__category-select-input"
                }
              }}
            >
              <MenuItem value="Other"><em>Other</em></MenuItem>
              <Divider/>
              {Object.keys(PRIMARY_CATEGORIES).map(c => (
                <MenuItem key={c} value={c}>
                  <ListItemIcon>
                    {PRIMARY_CATEGORIES[c]}
                  </ListItemIcon>
                  <ListItemText inset primary={c} />
                </MenuItem>
              ))}
              <Divider/>
              {Object.keys(SECONDARY_CATEGORIES).map(c => (
                <MenuItem key={c} value={c}>
                  <ListItemIcon>
                    {SECONDARY_CATEGORIES[c]}
                  </ListItemIcon>
                  <ListItemText inset primary={c} />
                </MenuItem>
              ))}
            </TextField>

            {/** Event Title */}
            <TextField
              required
              autoFocus
              id="eventTitle"
              label="Title"
              margin="normal"
              fullWidth
              onChange={this.handleEventTitleChange}
              value={this.state.eventTitle}
              error={titleError}
              helperText={titleError ? "This field is required" : ""}
            />

            {/** Date Picker */}
            <DatePicker
              day={this.state.eventDay}
              month={this.state.eventMonth}
              year={this.state.eventYear}
              onChange={this.handleEventDateChange} />

            {/** Event Description */}
            <TextField
              id="eventDesc"
              label="Description"
              multiline={true}
              margin="normal"
              className="timeline-add-event-desc"
              fullWidth
              onChange={this.handleEventDescChange}
              value={this.state.eventDesc}
              helperText="As long or short as you want!"
            />

            <TagInput 
                eventTags={this.state.eventTags}
                availableTags={_.difference(this.props.timeline.availableTags, this.state.eventTags)}
                onChange={this.handleEventTagChange} />

            {/** Image uploading and cropping */}
            <FormControl className="timeline-add-event-file-select">
              <Input
                type="file"
                margin="dense"
                onChange={this.handleEventImgChange}
              />
              <FormHelperText>
                Upload an image to remember the event
              </FormHelperText>
            </FormControl>
            {this.renderCropper()}

          </form>

          {/** Error toast */}
          <Snackbar
            anchorOrigin={{
              vertical: 'bottom',
              horizontal: 'left',
            }}
            open={this.state.snackbarOpen}
            autoHideDuration={2000}
            onClose={this.handleSnackbarClose}
            message={<span id="message-id">Please fill in the required fields</span>}
          />

        </Dialog>
      </div>
    );
  }
}

/* istanbul ignore next */
function mapStateToProps(state) {
  return {
    timeline: state.timeline,
  };
}

/* istanbul ignore next */
function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators({ ...actions, dataOperationSnackbarFailed, dataOperationSnackbarSucceeded }, dispatch)
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(AddEventForm);
