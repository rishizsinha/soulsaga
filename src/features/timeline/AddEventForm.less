@import '../../styles/mixins';

.timeline-add-event-fab {
  position: fixed;
  bottom: 32px;
  right: 32px;
}

.timeline-add-event-dialog__app-bar {
  position: relative;
}

.timeline-add-event-dialog__app-bar-text {
  flex: 1;
}

.timeline-add-event-form__container {
  margin: 32px;
}

.timeline-add-event__category-field {
  min-width: 175px;
}

.timeline-add-event__category-select-input {
  display: flex;
}

.timeline-add-event__category-select-menu {
  max-height: 350px;
}

.timeline-add-event-desc {
  overflow: hidden;
}

.timeline-add-event-file-select {
  margin: 32px 0px;
}

.timeline-add-event-img {
  max-width: 100%;
}