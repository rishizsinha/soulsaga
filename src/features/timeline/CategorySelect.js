import Divider from '@material-ui/core/Divider';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import MenuItem from '@material-ui/core/MenuItem';
import React, { Component } from 'react';
import TextField from '@material-ui/core/TextField';
import { PRIMARY_CATEGORIES, SECONDARY_CATEGORIES } from './eventiconmaps';

export default class CategorySelect extends Component {
  static propTypes = {

  };

  render() {
    return (
      <TextField
        select
        label="Category"
        value={this.props.category}
        onChange={this.props.onChange}
        className="timeline-category-select__field"
        margin="normal"
        SelectProps={{
          MenuProps: {
            className: "timeline-category-select__menu",
          },
        }}
        InputProps={{
          classes: {
            input: "timeline-category-select__input"
          }
        }}
      >
        <MenuItem value="Other"><em>Other</em></MenuItem>
        <Divider/>
        {Object.keys(PRIMARY_CATEGORIES).map(c => (
          <MenuItem key={c} value={c}>
            <ListItemIcon>
              {PRIMARY_CATEGORIES[c]}
            </ListItemIcon>
            <ListItemText inset primary={c} />
          </MenuItem>
        ))}
        <Divider/>
        {Object.keys(SECONDARY_CATEGORIES).map(c => (
          <MenuItem key={c} value={c}>
            <ListItemIcon>
              {SECONDARY_CATEGORIES[c]}
            </ListItemIcon>
            <ListItemText inset primary={c} />
          </MenuItem>
        ))}
      </TextField>
    );
  }
}
