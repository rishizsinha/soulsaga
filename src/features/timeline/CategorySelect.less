@import '../../styles/mixins';

.timeline-category-select {

}

.timeline-category-select__field {
  min-width: 175px;
  max-width: 175px;
  margin-right: 8px;
}

.timeline-category-select__input {
  display: flex;
}

.timeline-category-select__menu {
  max-height: 350px;
}
