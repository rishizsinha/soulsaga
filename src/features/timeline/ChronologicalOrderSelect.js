import MenuItem from '@material-ui/core/MenuItem';
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import TextField from '@material-ui/core/TextField';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import * as actions from './redux/actions';

export class ChronologicalOrderSelect extends Component {
  static propTypes = {
    timeline: PropTypes.object.isRequired,
    actions: PropTypes.object.isRequired,
  };

  changeOrder = e => this.props.actions.setChronoOrder(e.target.value);

  render() {
    return (
        <TextField
            select
            label="Chronological Order"
            value={this.props.timeline.eventOrdering}
            onChange={this.changeOrder}
            className="timeline-chronological-order-select"
            margin="normal"
          >
            <MenuItem value="forward">Forward</MenuItem>
            <MenuItem value="reverse">Reverse</MenuItem>
          </TextField>
    );
  }
}

/* istanbul ignore next */
function mapStateToProps(state) {
  return {
    timeline: state.timeline,
  };
}

/* istanbul ignore next */
function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators({ ...actions }, dispatch)
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ChronologicalOrderSelect);
