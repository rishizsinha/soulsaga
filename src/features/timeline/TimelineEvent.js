import {
  AccountIcon,
  AccountGroupIcon,
  AtomIcon,
  BookOpenIcon,
  BriefcaseIcon,
  CreationIcon,
  CurrencyUsdIcon,
  DumbbellIcon,
  EarthIcon,
  FoodIcon,
  HeartIcon,
  HomeIcon,
  HospitalIcon,
  LockIcon,
  PaletteIcon,
  PineTreeIcon,
  SchoolIcon,
  VectorCircleIcon} from 'mdi-react'
import Avatar from '@material-ui/core/Avatar';
import Card from '@material-ui/core/Card';
import CardHeader from '@material-ui/core/CardHeader';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import IconButton from '@material-ui/core/IconButton';
import Menu from '@material-ui/core/Menu';
import MenuItem from '@material-ui/core/MenuItem';
import MoreVertIcon from '@material-ui/icons/MoreVert';
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Typography from '@material-ui/core/Typography';
import { buildDateString } from './util';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import * as actions from './redux/actions';

const CATEGORY_ICON_MAP = {
  "Artistic": <PaletteIcon/>,
  "Community": <VectorCircleIcon/>,
  "Culinary": <FoodIcon/>,
  "Education": <SchoolIcon/>,
  "Family": <HomeIcon/>,
  "Financial": <CurrencyUsdIcon/>,
  "Fitness": <DumbbellIcon/>,
  "Intellectual": <BookOpenIcon/>,
  "Love/Romance": <HeartIcon/>,
  "Medical": <HospitalIcon/>,
  "Nature": <PineTreeIcon/>,
  "Personal": <AccountIcon/>,
  "Science": <AtomIcon/>,
  "Security": <LockIcon/>,
  "Social": <AccountGroupIcon/>,
  "Spiritual": <CreationIcon/>,
  "Travel": <EarthIcon/>,
  "Work/Career": <BriefcaseIcon/>,
  "Other": "R",
}

export class TimelineEvent extends Component {
  static propTypes = {
    timeline: PropTypes.object.isRequired,
    actions: PropTypes.object.isRequired,
  };

  state = {
    anchorEl: null,
  };

  handleClick = event => {
    this.setState({ anchorEl: event.currentTarget });
  };

  handleViewEvent = () => {
    this.props.actions.setViewingEvent(this.props.event);
  }

  handleClose = () => {
    this.setState({ anchorEl: null });
  };

  handleEdit = () => {
    this.handleClose();
    this.props.actions.setEditingEvent(this.props.event);
    this.props.actions.toggleAddEventForm(true);
  }

  handleDelete = () => {
    this.handleClose();
    this.props.actions.deleteEvent(
      this.props.event.id,
      this.props.event.ref);
  }

  renderImg() {
    if (this.props.event.img) {
      return (
        <CardMedia
          className="timeline-timeline-event__media"
          image={this.props.event.img}/>);
    }
  }

  renderDesc() {
    if (this.props.event.desc) {
      return (
        <CardContent className="timeline-timeline-event__desc-container">
          <Typography component="p" className="timeline-timeline-event__desc">
            {this.props.event.desc}
          </Typography>
        </CardContent>);
    }
  }

  renderMenu() {
    return (
      <Menu
        id="simple-menu"
        anchorEl={this.state.anchorEl}
        open={Boolean(this.state.anchorEl)}
        onClose={this.handleClose}
      >
        <MenuItem onClick={this.handleEdit}>Edit</MenuItem>
        <MenuItem onClick={this.handleDelete}>Delete</MenuItem>
      </Menu>
    );
  }

  render() {
    let dateString = buildDateString(
        this.props.event.day,
        this.props.event.month,
        this.props.event.year);
    return (
      <Card className="timeline-timeline-event__card" onClick={this.handleViewEvent}>
        <CardHeader
          avatar={
            <Avatar className="timeline-timeline-event__avatar">
              {CATEGORY_ICON_MAP[this.props.event.category]}
            </Avatar>
          }
          action={
            <IconButton onClick={this.handleClick}>
              <MoreVertIcon />
            </IconButton>
          }
          title={this.props.event.title}
          subheader={dateString}
        />
        {this.renderImg()}
        {this.renderDesc()}
        {this.renderMenu()}
      </Card>
    );
  }
}

/* istanbul ignore next */
function mapStateToProps(state) {
  return {
    timeline: state.timeline,
  };
}

/* istanbul ignore next */
function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators({ ...actions }, dispatch)
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(TimelineEvent);
