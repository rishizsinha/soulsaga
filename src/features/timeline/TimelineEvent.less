@import '../../styles/mixins';

.timeline-timeline-event__card {
  min-width: 320px;
  max-width: 320px;
  margin-right: 16px;
  margin-bottom: 16px;
}

.timeline-timeline-event__header-container {
  padding-bottom: 0;
}

.timeline-timeline-event__avatar {
  background-color: #E91E63;
}

.timeline-timeline-event__tag-container {
  margin: 4px 24px;
  display: flex;
  flex-wrap: nowrap;
  overflow-x: scroll;
}

.timeline-timeline-event__tag-container__end {
  margin-bottom: 24px;
}

.timeline-timeline-event__tag {
  margin-right: 4px;
  margin-bottom: 4px;
}

.timeline-timeline-event__tag > span > div {
  white-space: nowrap; 
  overflow: hidden;
  text-overflow: ellipsis;
  max-width: 80px;
}

.timeline-timeline-event__media {
 	height: 180px;
}

.timeline-timeline-event__desc-container {
  padding-bottom: 0px;
}

.timeline-timeline-event__desc {
  word-wrap: break-word;
  overflow: hidden;
  max-height: 80px;
  position: relative;
}

.timeline-timeline-event__desc:before {
    content:'';
    width:100%;
    height:100%;    
    position:absolute;
    left:0;
    top:0;
    background:linear-gradient(rgba(255, 255, 255, 0) 50px, white);
}

@media (min-width: 600px) {
  .timeline-timeline-event__expand-icon-container {
    margin-right: -8px;
  }
}