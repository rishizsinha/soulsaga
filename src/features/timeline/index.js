export { default as DefaultPage } from './DefaultPage';
export { default as TimelineEvent } from './TimelineEvent';
export { default as AddEventForm } from './AddEventForm';
export { default as DatePicker } from './DatePicker';
