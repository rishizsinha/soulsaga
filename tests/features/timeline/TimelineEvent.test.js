import React from 'react';
import { shallow } from 'enzyme';
import { TimelineEvent } from '../../../src/features/timeline/TimelineEvent';

describe('timeline/TimelineEvent', () => {
  it('renders node with correct class name', () => {
    const props = {
      timeline: {},
      actions: {},
    };
    const renderedComponent = shallow(
      <TimelineEvent {...props} />
    );

    expect(
      renderedComponent.find('.timeline-timeline-event').length
    ).toBe(1);
  });
});
