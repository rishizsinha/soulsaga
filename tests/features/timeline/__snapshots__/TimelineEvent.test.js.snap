// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`timeline/TimelineEvent renders proper structure based on event: category is transformed to icon 1`] = `
<div
  className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1 timeline-timeline-event__card"
  onClick={[Function]}
>
  <div
    className="MuiCardHeader-root-29"
  >
    <div
      className="MuiCardHeader-avatar-30"
    >
      <div
        className="MuiAvatar-root-35 MuiAvatar-colorDefault-36 timeline-timeline-event__avatar"
      >
        <svg
          className="mdi-icon "
          fill="currentColor"
          height={24}
          viewBox="0 0 24 24"
          width={24}
        >
          <path
            d="M10,20V14H14V20H19V12H22L12,3L2,12H5V20H10Z"
          />
        </svg>
      </div>
    </div>
    <div
      className="MuiCardHeader-content-32"
    >
      <span
        className="MuiTypography-root-38 MuiTypography-body2-46 MuiCardHeader-title-33"
      >
        testTitle
      </span>
      <span
        className="MuiTypography-root-38 MuiTypography-body2-46 MuiTypography-colorTextSecondary-71 MuiCardHeader-subheader-34"
      >
        April 2, 2010
      </span>
    </div>
  </div>
</div>
`;

exports[`timeline/TimelineEvent renders proper structure based on event: with tag and no desc/image, container-end class is appended 1`] = `
<div
  className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1 timeline-timeline-event__card"
  onClick={[Function]}
>
  <div
    className="MuiCardHeader-root-29 timeline-timeline-event__header-container"
  >
    <div
      className="MuiCardHeader-avatar-30"
    >
      <div
        className="MuiAvatar-root-35 MuiAvatar-colorDefault-36 timeline-timeline-event__avatar"
      >
        N
      </div>
    </div>
    <div
      className="MuiCardHeader-content-32"
    >
      <span
        className="MuiTypography-root-38 MuiTypography-body2-46 MuiCardHeader-title-33"
      >
        testTitle
      </span>
      <span
        className="MuiTypography-root-38 MuiTypography-body2-46 MuiTypography-colorTextSecondary-71 MuiCardHeader-subheader-34"
      >
        April 2, 2010
      </span>
    </div>
  </div>
  <div
    className="timeline-timeline-event__tag-container timeline-timeline-event__tag-container__end"
  >
    <div
      className="MuiChip-root-77 timeline-timeline-event__tag"
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex={-1}
    >
      <span
        className="MuiChip-label-96"
      >
        <div>
          testTag1
        </div>
      </span>
    </div>
    <div
      className="MuiChip-root-77 timeline-timeline-event__tag"
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex={-1}
    >
      <span
        className="MuiChip-label-96"
      >
        <div>
          testTag2
        </div>
      </span>
    </div>
  </div>
</div>
`;

exports[`timeline/TimelineEvent renders proper structure based on event: with title and description, has CardHeader and CardContent 1`] = `
<div
  className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1 timeline-timeline-event__card"
  onClick={[Function]}
>
  <div
    className="MuiCardHeader-root-29"
  >
    <div
      className="MuiCardHeader-avatar-30"
    >
      <div
        className="MuiAvatar-root-35 MuiAvatar-colorDefault-36 timeline-timeline-event__avatar"
      >
        N
      </div>
    </div>
    <div
      className="MuiCardHeader-content-32"
    >
      <span
        className="MuiTypography-root-38 MuiTypography-body2-46 MuiCardHeader-title-33"
      >
        testTitle
      </span>
      <span
        className="MuiTypography-root-38 MuiTypography-body2-46 MuiTypography-colorTextSecondary-71 MuiCardHeader-subheader-34"
      >
        April 2, 2010
      </span>
    </div>
  </div>
  <div
    className="MuiCardContent-root-74 timeline-timeline-event__desc-container"
  >
    <p
      className="MuiTypography-root-38 MuiTypography-body1-47 timeline-timeline-event__desc"
    >
      testDescription
    </p>
  </div>
</div>
`;

exports[`timeline/TimelineEvent renders proper structure based on event: with title and image, has CardHeader and CardMedia 1`] = `
<div
  className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1 timeline-timeline-event__card"
  onClick={[Function]}
>
  <div
    className="MuiCardHeader-root-29"
  >
    <div
      className="MuiCardHeader-avatar-30"
    >
      <div
        className="MuiAvatar-root-35 MuiAvatar-colorDefault-36 timeline-timeline-event__avatar"
      >
        N
      </div>
    </div>
    <div
      className="MuiCardHeader-content-32"
    >
      <span
        className="MuiTypography-root-38 MuiTypography-body2-46 MuiCardHeader-title-33"
      >
        testTitle
      </span>
      <span
        className="MuiTypography-root-38 MuiTypography-body2-46 MuiTypography-colorTextSecondary-71 MuiCardHeader-subheader-34"
      >
        April 2, 2010
      </span>
    </div>
  </div>
  <div
    className="MuiCardMedia-root-75 timeline-timeline-event__media"
    style={
      Object {
        "backgroundImage": "url(\\"testImg\\")",
      }
    }
  />
</div>
`;

exports[`timeline/TimelineEvent renders proper structure based on event: with title, description and image, has CardHeader, CardContent
        and CardMedia 1`] = `
<div
  className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1 timeline-timeline-event__card"
  onClick={[Function]}
>
  <div
    className="MuiCardHeader-root-29"
  >
    <div
      className="MuiCardHeader-avatar-30"
    >
      <div
        className="MuiAvatar-root-35 MuiAvatar-colorDefault-36 timeline-timeline-event__avatar"
      >
        N
      </div>
    </div>
    <div
      className="MuiCardHeader-content-32"
    >
      <span
        className="MuiTypography-root-38 MuiTypography-body2-46 MuiCardHeader-title-33"
      >
        testTitle
      </span>
      <span
        className="MuiTypography-root-38 MuiTypography-body2-46 MuiTypography-colorTextSecondary-71 MuiCardHeader-subheader-34"
      >
        April 2, 2010
      </span>
    </div>
  </div>
  <div
    className="MuiCardMedia-root-75 timeline-timeline-event__media"
    style={
      Object {
        "backgroundImage": "url(\\"testImg\\")",
      }
    }
  />
  <div
    className="MuiCardContent-root-74 timeline-timeline-event__desc-container"
  >
    <p
      className="MuiTypography-root-38 MuiTypography-body1-47 timeline-timeline-event__desc"
    >
      testDescription
    </p>
  </div>
</div>
`;

exports[`timeline/TimelineEvent renders proper structure based on event: with title, only has CardHeader 1`] = `
<div
  className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1 timeline-timeline-event__card"
  onClick={[Function]}
>
  <div
    className="MuiCardHeader-root-29"
  >
    <div
      className="MuiCardHeader-avatar-30"
    >
      <div
        className="MuiAvatar-root-35 MuiAvatar-colorDefault-36 timeline-timeline-event__avatar"
      >
        N
      </div>
    </div>
    <div
      className="MuiCardHeader-content-32"
    >
      <span
        className="MuiTypography-root-38 MuiTypography-body2-46 MuiCardHeader-title-33"
      >
        testTitle
      </span>
      <span
        className="MuiTypography-root-38 MuiTypography-body2-46 MuiTypography-colorTextSecondary-71 MuiCardHeader-subheader-34"
      >
        April 2, 2010
      </span>
    </div>
  </div>
</div>
`;

exports[`timeline/TimelineEvent renders proper structure based on event: with title, tags, description and image, has CardHeader, Chips,
        CardContent and CardMedia 1`] = `
<div
  className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1 timeline-timeline-event__card"
  onClick={[Function]}
>
  <div
    className="MuiCardHeader-root-29 timeline-timeline-event__header-container"
  >
    <div
      className="MuiCardHeader-avatar-30"
    >
      <div
        className="MuiAvatar-root-35 MuiAvatar-colorDefault-36 timeline-timeline-event__avatar"
      >
        N
      </div>
    </div>
    <div
      className="MuiCardHeader-content-32"
    >
      <span
        className="MuiTypography-root-38 MuiTypography-body2-46 MuiCardHeader-title-33"
      >
        testTitle
      </span>
      <span
        className="MuiTypography-root-38 MuiTypography-body2-46 MuiTypography-colorTextSecondary-71 MuiCardHeader-subheader-34"
      >
        April 2, 2010
      </span>
    </div>
  </div>
  <div
    className="timeline-timeline-event__tag-container"
  >
    <div
      className="MuiChip-root-77 timeline-timeline-event__tag"
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex={-1}
    >
      <span
        className="MuiChip-label-96"
      >
        <div>
          testTag1
        </div>
      </span>
    </div>
    <div
      className="MuiChip-root-77 timeline-timeline-event__tag"
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex={-1}
    >
      <span
        className="MuiChip-label-96"
      >
        <div>
          testTag2
        </div>
      </span>
    </div>
  </div>
  <div
    className="MuiCardMedia-root-75 timeline-timeline-event__media"
    style={
      Object {
        "backgroundImage": "url(\\"testImg\\")",
      }
    }
  />
  <div
    className="MuiCardContent-root-74 timeline-timeline-event__desc-container"
  >
    <p
      className="MuiTypography-root-38 MuiTypography-body1-47 timeline-timeline-event__desc"
    >
      testDescription
    </p>
  </div>
</div>
`;
